File: src/org/symbolmath/scanner/TokenType.java
package org.symbolmath.scanner;

public enum TokenType {
  PLUS("+"),
  MINUS("-"),
  TIMES("*"),
  DIVISION("/"),
  POWER("^"),
  LPAREN("("),
  RPAREN(")"),
  EQUALS("="),
  INTEGER(),
  FLOAT(),
  IDENTIFIER(),
  WHITESPACE(),
  EOF();

  private String text;

  TokenType() {
  }

  TokenType(String text) {
    this.text = text;
  }

  public String getText() {
    return text;
  }
}

File: src/main/java/org/symbolmath/scanner/TokenType.kt
package org.symbolmath.scanner

enum class TokenType {
    PLUS("+"), MINUS("-"), TIMES("*"), DIVISION("/"), POWER("^"),
    LPAREN("("), RPAREN(")"),
    EQUALS("="),
    INTEGER, FLOAT, IDENTIFIER, WHITESPACE,
    EOF;

    var text: String? = null
        private set

    constructor()

    constructor(text: String) {
        this.text = text
    }
}